collect_app/src/androidTest/java/org/odk/collect/android/regression/FillBlankFormTest.java

```diff
@@ -340,79 +353,79 @@ public void values_ShouldBeRandom() {
     public void app_ShouldNotCrash() {
 
         //TestCase32
-        MainMenu.startBlankForm("g6Error");
-        FormEntry.checkIsStringDisplayed(R.string.error_occured);
-        FormEntry.clickOk();
-        FormEntry.swipeToNextQuestion();
-        FormEntry.clickSaveAndExit();
-        FormEntry.checkIsToastWithMessageDisplayes("Form successfully saved!", main);
-
-        MainMenu.startBlankForm("g6Error2");
-        FormEntry.putText("bla");
-        FormEntry.swipeToNextQuestion();
-        FormEntry.checkIsStringDisplayed(R.string.error_occured);
-        FormEntry.clickOk();
-        FormEntry.swipeToNextQuestion();
-        FormEntry.putText("ble");
-        FormEntry.swipeToNextQuestion();
-        FormEntry.clickSaveAndExit();
-        FormEntry.checkIsToastWithMessageDisplayes("Form successfully saved!", main);
-
-        MainMenu.startBlankForm("emptyGroupFieldList");
-        FormEntry.clickSaveAndExit();
-        FormEntry.checkIsToastWithMessageDisplayes("Form successfully saved!", main);
-
-        MainMenu.startBlankForm("emptyGroupFieldList2");
-        FormEntry.putText("nana");
-        FormEntry.swipeToNextQuestion();
-        FormEntry.clickSaveAndExit();
-        FormEntry.checkIsToastWithMessageDisplayes("Form successfully saved!", main);
+        new MainMenuPage(main)
+                .startBlankFormWithError("g6Error")
+                .clickOK(new FormEntryPage("g6Error", main))
+                .swipeToNextQuestion()
+                .clickSaveAndExit()
+                .checkIsToastWithMessageDisplayed("Form successfully saved!");
+
+        new MainMenuPage(main).startBlankForm("g6Error2")
+                .putText("bla")
+                .swipeToNextQuestionWithError()
```

Comment:
I've adopted this pattern of giving actions a `WithError` suffix and returning an `ErrorDialog` when we expect there to be an error. A lot of other ways to handle this but let's see how this evolves.